core/src/main/java/org/apache/calcite/sql/fun/SqlLibraryOperators.java

```diff
@@ -360,4 +367,23 @@ private SqlLibraryOperators() {
   public static final SqlOperator INFIX_CAST =
       new SqlCastOperator();
 
+  @LibraryOperator(libraries = {BIGQUERY, HIVE, SPARK})
+  public static final SqlFunction IF =
+      new SqlFunction(
+          "IF",
+          SqlKind.IF,
+          ReturnTypes.ARG1_NULLABLE,
```

Comment:
Should it be `ReturnTypes.cascade(ReturnTypes.LEAST_RESTRICTIVE, SqlTypeTransforms.LEAST_NULLABLE)` ?